scipy/linalg/basic.py

```diff
@@ -23,21 +23,43 @@
 
 # Linear equations
 def solve(a, b, sym_pos=False, lower=False, overwrite_a=False,
-          overwrite_b=False, debug=False, check_finite=True):
+          overwrite_b=False, check_finite=True, assume_a='gen',
```

Comment:
I think better separate the backwards incompatible change from this PR, then we don't have to think about it now.